### ISSUE TITLE
Embedding dimension transformation in experiments of Section 3

### ISSUE THREAD

**Issue by ZMHH-H** [fletcherjiang/LLMEPET#5]
Hi, thanks for your great job. I'm curious how you transform the CLIP embedding dimension to match the LLAM in the experiments of Section 3. If a linear layer is used, how is the layer trained? or is it just randomly initialized?

**Comment by fletcherjiang** [comment]
In our setup, we use a linear layer to transform the CLIP embedding dimension to match the LLAM. Specifically, we utilize two linear mappings, `llama_dim_mapper1` and `llama_dim_mapper2`, to project the dimensions. The first mapper transforms the `d_model` dimension to `dim` (4096), and the second mapper projects it back to `d_model`. Both layers are trainable by default to fine-tune the embeddings, allowing adaptation within the LLaMA Transformer context. However, we freeze the parameters of the LLaMA model after loading the pre-trained weights from the checkpoint, meaning only the mapper layers are updated during training, which helps integrate the CLIP embeddings effectively.

I hope the answer answers your query~

**Comment by ZMHH-H** [comment]
Thanks for your reply! Does this mean the same model is used for pre-experiment (Section 3) and your main method (Section 5)?

**Comment by fletcherjiang** [comment]
Yes, the frozen llama models are all the same, and all go through a linear layer to be converted to the same dimensions.
